compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -117,12 +117,35 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
     }
     else
     {
-      auto shape_node = loco::must_cast<luci::CircleNode *>(node->shape());
-      assert(shape_node->rank() == 1);
-      // shape_node tensor values will provide new shape, like [2, 3, 4]
-      auto num_elements = shape_node->dim(0).value(); // above example will give 3
-      shape_by_input.rank(num_elements);
-      is_static_shape = false;
+      // NOTE assumption is that `shape` and `newShape` having same value.
+      // for non-existing `shape`, we can use `newShape` if it's valid
+      auto new_shape = node->newShape();
+      auto rank = new_shape->rank();
```

Comment:
Better to check new_shape is not nullptr to avoid svace complaint.

Reply:
new_shape is ptr of member and never nullptr :)

Reply:
if svace complains... I'd liked like to complain to svace too :)